### PR TITLE
XOR-252 [FEAT] Add "Select all" functionality for checkboxes

### PR DESCRIPTION
**Pull request by anshukdutt**
### Product areas affected
Fliplet-Form builder

### What does this PR do?
Ensures to add selectAll functionality for checkboxes in via form builder configuration & JS API
### JIRA ticket
[JIRA](https://weboo.atlassian.net/browse/XOR-252)

### Result
Add any relevant sample input/output, screenshots or screen recordings.

[Screen Recording](https://api.media.atlassian.com/file/6fbff0a6-b23e-42fe-a767-9f9190e79077/artifact/video_1280.mp4/binary?client=435f5956-a841-4ea9-80ca-d94a2672bba6&collection=&max-age=2592000&token=eyJhbGciOiJIUzI1NiJ9.eyJpc3MiOiI0MzVmNTk1Ni1hODQxLTRlYTktODBjYS1kOTRhMjY3MmJiYTYiLCJhY2Nlc3MiOnsidXJuOmZpbGVzdG9yZTpmaWxlOmI1OTJmNGM1LTZmZDYtNGU3Zi1hZTYwLTJlZDg1NjI3MWZiOSI6WyJyZWFkIl0sInVybjpmaWxlc3RvcmU6ZmlsZTplNjRhNzJlZi1jY2VlLTQzM2UtYWM3Yi0wNmQzNDA5ZWNhOGMiOlsicmVhZCJdLCJ1cm46ZmlsZXN0b3JlOmZpbGU6OTZhZDFmNzgtMWViMi00YzBkLWEzZmYtYTc0MjRmNmYwNzY5IjpbInJlYWQiXSwidXJuOmZpbGVzdG9yZTpmaWxlOjZmYmZmMGE2LWIyM2UtNDJmZS1hNzY3LTlmOTE5MGU3OTA3NyI6WyJyZWFkIl0sInVybjpmaWxlc3RvcmU6ZmlsZTo1OWM3MGJmZS02YTIyLTQ4ODAtYmNmYS1lMGU5NDYyMTEwMWUiOlsicmVhZCJdLCJ1cm46ZmlsZXN0b3JlOmZpbGU6MzMxZWM1ODItNzdlYS00NzFmLWJlZmMtYWIyMzAxYmIyMTQwIjpbInJlYWQiXSwidXJuOmZpbGVzdG9yZTpmaWxlOjBiYjYzODhjLTRiNjMtNDhlYi1hY2E3LWQ3YTIwY2RlZGQzMyI6WyJyZWFkIl0sInVybjpmaWxlc3RvcmU6ZmlsZTpjNDdjMmU5Yy1kYjAyLTRlNDUtYmFmOC0wYjdlOWEzMzkzZjYiOlsicmVhZCJdLCJ1cm46ZmlsZXN0b3JlOmZpbGU6MDk1MzYxNGMtMmY0ZS00ZGU3LThlOWYtYjBlNDY4ZmFlNGUxIjpbInJlYWQiXSwidXJuOmZpbGVzdG9yZTpmaWxlOmQ0Y2IzOWIwLWQwOTItNDc1Ny05YTQ3LWYxMGJmNThkOTZhOSI6WyJyZWFkIl19LCJleHAiOjE2NTcxODYzMzQsIm5iZiI6MTY1NzE4NTQzNH0.1A9LE9SNpvrnWZsmX6X4hLiF-_c595sZrnBiy-HKtUg)

### Deployment instructions
None

### Author concerns
Script used to create JS API: 
Fliplet.FormBuilder.get().then(function(form) {
  form.fields.set([
    {
      _type: 'flCheckbox',
      selectAll: true,
      options: [{ label: 'zero' }, { label: 'two' }, { label: 'one' }]
    }
  ]);
});

### CODE REVIEW

File: js/components/checkbox.js [hunk 1]
```diff
@@ -63,12 +97,24 @@ Fliplet.FormBuilder.field('checkbox', {
...
+        $vm.value.push(val);
+      });
+      } 
       this.updateValue();
     }
   },
   created: function() {
     var $vm = this;
-
+    console.log('Value');
```
Comment (yogeshmsharma-xoriant) [inline js/components/checkbox.js]: pls remove this

File: js/components/checkbox.js [hunk 2]
```diff
@@ -83,14 +129,38 @@ Fliplet.FormBuilder.field('checkbox', {
       });
 
       this.value = selectedOptions.length ? _.uniqWith(this.value, _.isEqual) : [];
+      console.log(this.value);
```
Comment (yogeshmsharma-xoriant) [inline js/components/checkbox.js]: pls remove all console.log

File: templates/components/checkbox.build.hbs [hunk 3]
```diff
@@ -1,5 +1,28 @@
 <template>
   <div v-if="description" class="help-block description">\{{ description }}</div>
+    <template v-if="isSelectAll">
+            <div class="checkbox checkbox-icon">
+       <input
+          type="checkbox"
+          :id="name + '-' + 'a'"
```
Comment (yogeshmsharma-xoriant) [inline templates/components/checkbox.build.hbs]: what is this for?

**Review by yogeshmsharma-xoriant — COMMENTED** [review]
pls do minor changes

File: js/components/checkbox.js [hunk 4]
```diff
@@ -48,11 +62,31 @@ Fliplet.FormBuilder.field('checkbox', {
           return (option.label || option.id) === val;
         });
       });
-
       this.highlightError();
 
       this.$emit('_input', this.name, ordered);
     },
+    updateCheckValue: function() {
```
Comment (tonytlwu) [inline js/components/checkbox.js:69]: I don't believe this function is necessary.

```suggestion
```

File: js/components/checkbox.js [hunk 5]
```diff
@@ -25,6 +25,20 @@ Fliplet.FormBuilder.field('checkbox', {
           }
         ];
       }
+    },
+    isSelectAll: {
```
Comment (tonytlwu) [inline js/components/checkbox.js:29]: This should not be necessary to render or manage the **Select all** option.

```suggestion
```

File: templates/components/checkbox.build.hbs [hunk 6]
```diff
@@ -1,6 +1,30 @@
 <template>
   <div v-if="description" class="help-block description">\{{ description }}</div>
 
+  <template v-if="selectAll">
+    <div class="checkbox checkbox-icon">
+      <input
+        type="checkbox"
+        :id="name + '-' + 'sa'"
+        :name="name"
```
Comment (tonytlwu) [inline templates/components/checkbox.build.hbs:9]: I don't believe a `name` attribute is necessary.

```suggestion
```

**Review by tonytlwu — CHANGES_REQUESTED** [review]

**Comment by tonytlwu** [comment]
Closing the PR now that it's implemented through https://github.com/Fliplet/fliplet-widget-form-builder/pull/454

**Pull request closed without merging**